pkg/prometheus/rules.go

```diff
@@ -355,11 +355,13 @@ func ValidateRule(promRule monitoringv1.PrometheusRuleSpec) []error {
 		if group.PartialResponseStrategy == "" {
 			continue
 		}
+		// TODO(slashpai): Remove this validation after v0.65 since this is handled at CRD level
```

Comment:
I'm not sure why the operator checks the PartialResponseStrategy value since it is reset anyway...

Reply:
I think that was added to check for ThanosRuler but I dont see this func is called for that. I removed the check